src/orange/OrangeTrackView.hh

```diff
@@ -368,15 +469,23 @@ CELER_FUNCTION real_type OrangeTrackView::find_safety()
  */
 CELER_FUNCTION void OrangeTrackView::move_to_boundary()
 {
-    CELER_EXPECT(states_.boundary[thread_] != BoundaryResult::reentrant);
+    LevelStateAccessor lsa(&states_, thread_, states_.level[thread_]);
+
+    CELER_EXPECT(lsa.boundary() != BoundaryResult::reentrant);
     CELER_EXPECT(this->has_next_step());
     CELER_EXPECT(next_surface_);
 
     // Physically move next step
-    axpy(next_step_, states_.dir[thread_], &states_.pos[thread_]);
+    // axpy(next_step_, states_.dir[thread_], &states_.pos[thread_]);
+
+    auto pos = lsa.pos();
+    axpy(next_step_, lsa.dir(), &pos);
+    lsa.set_pos(pos);
```

Comment:
I wonder if the `LSA` class should return mutable references rather than have setters? That would allow you to write
```c++
axpy(next_step_, lsa.dir(), lsa.pos());
```
although the disadvantage is losing the flexibility to restructure the pos/dir data (which I think is OK for now).

Reply:
This is how I was trying to do it originally, but I couldn't get the types right w.r.t. `const` and `Collections`. I decided to punt first and revisit it.

Reply:
The constness with collections is a bit wacky... so we may want to pair program on this a bit.